### PR TITLE
Feature/try archive

### PR DESCRIPTION
**Pull request by bernardoezequias**

### CODE REVIEW

File: src/main/java/saps/archiver/core/ArchiverMain.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package saps.archiver.core;
+
+//TODO: missing imports
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/ArchiverMain.java]: could we remove this TODO?

File: src/main/java/saps/archiver/core/ArchiverMain.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+public class ArchiverMain {
+
+    public static void main(String[] args) throws Exception {
+        Properties properties = new Properties();
+        FileInputStream input = new FileInputStream([0]);
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/ArchiverMain.java]: `[0]`? nao deveria ser `args[0]`

File: src/main/java/saps/archiver/core/ArchiverMain.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        properties.load(input);
+
+        Archiver fetcher = createArchiver(properties);
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/ArchiverMain.java]: archiver or fetcher?

File: src/main/java/saps/archiver/core/ArchiverMain.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        while (true) {
+            fetcher.gc();
+            fetcher.archive();
+            Thread.sleep(fetcher.getDelayMilis());
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/ArchiverMain.java]: it is odd to have this getDelayMilis. maybe, this config should be read here in the main

File: src/main/java/saps/archiver/core/ArchiverMain.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        }
+    }
+
+    private static Archiver createArchiver(Properties properties) throws Exception {
+        PermanentStorage permanentStorage = createPermanentStorage(properties);
+        Catalog catalog = new JDBCCatalog(properties);
+        Archiver archiver = new DefaultArchiver(properties, catalog, permanentStorage);
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/ArchiverMain.java]: there is no need to declare this archiver variable here

File: src/main/java/saps/archiver/core/ArchiverMain.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        return archiver;
+    }
+
+    private static PermanentStorage createPermanentStorage(Properties properties) throws Exception {
+        String permanentStorageType = properties.getProperty(SapsPropertiesConstants.SAPS_PERMANENT_STORAGE_TYPE);
+        return new NfsPermanentStorage(properties);
+        throw new SapsException("Failed to recognize type of permanent storage");
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/ArchiverMain.java]: this line 30 seems wrong

File: src/main/java/saps/archiver/core/DefaultArchiver.java [hunk 7]
```diff
@@ -1,6 +1,132 @@
...
+import saps.archiver.interfaces.*;
+import saps.archiver.core.exceptions.*;
+
+import java.io.IOException;
+import java.util.List;
+
+import java.io.File;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+//TODO imports missing
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/DefaultArchiver.java]: could we remove this TODO?

File: src/main/java/saps/archiver/core/DefaultArchiver.java [hunk 8]
```diff
@@ -1,6 +1,132 @@
...
+
+  private static final Logger LOGGER = Logger.getLogger(Archiver.class);
+
+  public DefaultArchiver(
+    Properties properties,
+    Catalog catalog,
+    PermanentStorage permanentStorage) throws WrongConfigurationException {
+
+    if (!checkProperties(properties)) {
+    	throw new WrongConfigurationException("Error on validate the file. Missing properties for start Saps Controller.");
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/DefaultArchiver.java]: I think there is already a java sdk exception to this

File: src/main/java/saps/archiver/core/DefaultArchiver.java [hunk 9]
```diff
@@ -1,6 +1,132 @@
...
+  public void archive() {
+
+    List <SapsImage> tasksToArchive = CatalogUtils.getTasks(catalog, ImageTaskState.FINISHED);
+    for (SapsImage task: tasksToArchive) {
+      updateTaskState(task, ImageTaskState.ARCHIVING);
+      if (archive(task)) {
+        updateTaskState(task, ImageTaskState.ARCHIVED); 
+      } else {
+        updateTaskState(task, ImageTaskState.FAILED);
+      } deleteTempData(task);
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/DefaultArchiver.java]: put the delete call (line 51) in line 52

File: src/main/java/saps/archiver/core/ArchiverMain.java [hunk 10]
```diff
@@ -0,0 +1,32 @@
+package saps.archiver.core;
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/ArchiverMain.java:1]: check code identation i

File: src/main/java/saps/archiver/core/DefaultArchiver.java [hunk 11]
```diff
@@ -1,6 +1,132 @@
...
+  };
+
+  private boolean checkProperties(Properties properties) {
+    String[] propertiesSet = {
+      SapsPropertiesConstants.IMAGE_DATASTORE_IP,
+      SapsPropertiesConstants.IMAGE_DATASTORE_PORT,
+      SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_GARBAGE_COLLECTOR,
+      SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_ARCHIVER,
+      SapsPropertiesConstants.SAPS_TEMP_STORAGE_PATH,
+      SapsPropertiesConstants.SAPS_PERMANENT_STORAGE_TYPE
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/DefaultArchiver.java]: are we using all these constants?

File: src/main/java/saps/archiver/core/DefaultArchiver.java [hunk 12]
```diff
@@ -1,6 +1,132 @@
...
+  /**
+   * It deletes the data generated by {@code SapsImage} in the temp storage.
+   *
+   * @param task {@code SapsImage}
+   */
+  private void deleteTempData(SapsImage task) {    
+    String taskDirPath = tempStoragePath + File.separator + task.getTaskId();
+    
+    File taskDir = new File(taskDirPath);
+    if (taskDir.exists() && taskDir.isDirectory()) {
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/DefaultArchiver.java]: maybe remove this call to a std java method?

File: src/main/java/saps/archiver/core/DefaultArchiver.java [hunk 13]
```diff
@@ -1,6 +1,132 @@
...
+        FileUtils.deleteDirectory(taskDir);
+      } catch (IOException e) {
+        LOGGER.error("Error while delete task [" + task.getTaskId() + "] files from disk: ", e);
+      }
+    } 
+  }
 
-    public void archive();
+  /**
+   * It returns the maximum value between garbage collector and archiver delay constants
```
Comment (thiagomanel) [inline src/main/java/saps/archiver/core/DefaultArchiver.java]: this method shoudl be removed

**Review by thiagomanel — CHANGES_REQUESTED** [review]
many comments but I need tests before accepting the PR

**Review by thiagomanel — CHANGES_REQUESTED** [review]
I need to be able to run the test in the command line `mvn test`
also, I think the `pom.xml` has many unused dependencies. please, check this and remove any unused dependency

**Comment by bernardoezequias** [comment]
> I need to be able to run the test in the command line `mvn test` also, I think the `pom.xml` has many unused dependencies. please, check this and remove any unused dependency

done

**Review by thiagomanel — APPROVED** [review]
lgtm

**Review by jhenriqueax — APPROVED** [review]

**Pull request merged**